binder/src/main/java/io/grpc/binder/BinderServerBuilder.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.binder;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import android.app.Service;
+import android.os.IBinder;
+import com.google.common.base.Supplier;
+import io.grpc.CompressorRegistry;
+import io.grpc.DecompressorRegistry;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.ServerStreamTracer;
+import io.grpc.binder.internal.BinderServer;
+import io.grpc.binder.internal.BinderTransportSecurity;
+import io.grpc.internal.AbstractServerImplBuilder;
+import io.grpc.internal.FixedObjectPool;
+import io.grpc.internal.GrpcUtil;
+import io.grpc.internal.InternalServer;
+import io.grpc.internal.ServerImplBuilder;
+import io.grpc.internal.ServerImplBuilder.ClientTransportServersBuilder;
+import io.grpc.internal.ObjectPool;
+import io.grpc.internal.SharedResourcePool;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import javax.annotation.Nullable;
+
+/**
+ * Builder for a server that services requests from an Android Service.
+ */
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/8022")
+public final class BinderServerBuilder
+    extends AbstractServerImplBuilder<BinderServerBuilder> {
```

Comment:
We'll want to swap this over to ForwardingServerBuilder. AbstractServerImplBuilder is a copy of ForwardingServerBuilder to preserve ABI. I don't care if it happens now or later.

Reply:
Done.